### PR TITLE
Upgrade to new library

### PR DESCRIPTION
**Pull request by licarth**
- [x] Upgrades `remix` & adapt types
- [x] Uses latest version of @howmuchgreen/howmuchcarbon
- [x] Updates tsconfig with recommended values  
	CleanShot 2022-08-04 at 18 15 03@2x

TODO before merging
- [ ] Replace lib with version when [this PR](https://github.com/howmuchgreen/howmuchcarbon/pull/10) is merged.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,20 +11,20 @@
     "start": "remix-serve build"
   },
   "dependencies": {
-    "@howmuchgreen/howmuchcarbon": "../howmuchcarbon/library/published",
+    "@howmuchgreen/howmuchcarbon": "../howmuchcarbon/library/dist/published",
```
Comment (licarth) [inline package.json:14]: To be replaced when lib is published

File: app/routes/api/$q/index.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {
```
Comment (NTag) [inline app/routes/api/$q/index.tsx:1]: I don't think this file should be a `.tsx`

**Review by NTag — COMMENTED** [review]
Nice!

File: app/routes/api/$q/index.tsx (hunk 2 above)
Comment (licarth) [reply app/routes/api/$q/index.tsx:1]: Yup this is a remains from when I was trying to debug it and see whether extension made a difference

**Review by NTag — APPROVED** [review]

**Pull request merged**